frontend/src/store/RealmSlice.ts

```diff
@@ -0,0 +1,28 @@
+// SPDX-FileCopyrightText: 2024 German Aerospace Center (DLR) and CISPA Helmholtz Center for Information Security
+// SPDX-License-Identifier: Apache-2.0
+
+import {createSlice, PayloadAction} from '@reduxjs/toolkit';
+
+export interface Realm {
+  name: string;
+}
+
+const initialState: Realm = {
+  name: localStorage.getItem('realm') || '',
+};
+
+export const RealmSlice = createSlice({
+  name: 'Realm',
+  initialState,
+  reducers: {
+    setRealm(state, action: PayloadAction<string>) {
+      // store realm in local storage
+      // redirects start new the browser sessions i.e. store state will be reset
+      localStorage.setItem('realm', action.payload);
+      state.name = action.payload;
+    },
+  },
+});
```

Comment:
We already have a feature to persist data, so you don't have to do this manually. In the `index.ts` is the `persistConfig`. You can just add your realm slice to the whitelist and should automatically get persisted.

```suggestion
export interface Realm {
  name: string;
}

const initialState: Realm = {
  name: '',
};

export const RealmSlice = createSlice({
  name: 'Realm',
  initialState,
  reducers: {
    setRealm(state, action: PayloadAction<string>) {
      state.name = action.payload;
    },
  },
});
```